controllers/create_metrics_monitor.go

```diff
@@ -0,0 +1,159 @@
+// Copyright Contributors to the Open Cluster Management project
+package controllers
+
+import (
+	"context"
+
+	monitorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	searchv1alpha1 "github.com/stolostron/search-v2-operator/api/v1alpha1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *SearchReconciler) createMetricsRole(ctx context.Context,
+	crole *rbacv1.Role,
+) (*reconcile.Result, error) {
+
+	found := &rbacv1.Role{}
+	err := r.Get(ctx, types.NamespacedName{
+		Name:      crole.Name,
+		Namespace: crole.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+		err = r.Create(ctx, crole)
+		if err != nil {
+			log.Error(err, "Could not create role "+crole.Name)
+			return &reconcile.Result{}, err
+		}
+		log.Info("Created role" + crole.Name)
+		log.V(9).Info("Created  role ", "name", crole)
+	}
+	return nil, nil
+}
+
+func (r *SearchReconciler) createMetricsRoleBinding(ctx context.Context,
+	rolebinding *rbacv1.RoleBinding,
+) (*reconcile.Result, error) {
+
+	found := &rbacv1.RoleBinding{}
+	err := r.Get(ctx, types.NamespacedName{
+		Name:      rolebinding.Name,
+		Namespace: rolebinding.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+		err = r.Create(ctx, rolebinding)
+		if err != nil {
+			log.Error(err, "Could not create rolebinding"+rolebinding.Name)
+			return &reconcile.Result{}, err
+		}
+		log.Info("Created rolebinding" + rolebinding.Name)
+		log.V(2).Info("Created rolebinding ", "name", rolebinding)
+	}
+	return nil, nil
+}
+
+func (r *SearchReconciler) MetricsRole(instance *searchv1alpha1.Search) *rbacv1.Role {
+	cr := &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Role",
+			APIVersion: rbacv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "search-metrics-monitor",
+			Namespace: instance.GetNamespace(),
+		},
+		Rules: getRules(),
+	}
+	err := controllerutil.SetControllerReference(instance, cr, r.Scheme)
+	if err != nil {
+		log.Info("Could not set control for Role ", "name", "search-metrics-monitor")
```

Comment:
Should this be logged as a warning or error? Similar pattern in the functions below.

Reply:
I don't think so. For pre-existing functions, it is not treated as an error.